authFetch.js

```diff
@@ -0,0 +1,29 @@
+var HMAC_SECRET = require('./secrets').HMAC_SECRET;
+var jwtUtil = require('./JWT/jwtUtil');
+var fetch = require('node-fetch');
+
+function authFetch (url, data) {
+  data = addAuthToken(data);
+  return fetch(url, data);
+}
+
+function addAuthToken(data) {
+  if (!data) {
+    data = {};
+  }
+  if (!data.headers) {
+    data.headers = {};
+  }
```

Comment:
Why not
`
if (!data || !data.headers) {
    data = {
        headers: {}
    };
}
`